customer-backend/src/main/java/io/eventuate/examples/tram/ordersandcustomers/customers/service/CustomerOptimisticLockingDecorator.java

```diff
@@ -0,0 +1,28 @@
+package io.eventuate.examples.tram.ordersandcustomers.customers.service;
+
+import io.eventuate.tram.consumer.common.MessageHandlerDecorator;
+import io.eventuate.tram.consumer.common.MessageHandlerDecoratorChain;
+import io.eventuate.tram.consumer.common.SubscriberIdAndMessage;
+import org.springframework.core.Ordered;
+import org.springframework.orm.ObjectOptimisticLockingFailureException;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+
+import javax.transaction.Transactional;
+
+@Transactional
+public class CustomerOptimisticLockingDecorator implements MessageHandlerDecorator, Ordered {
```

Comment:
This should be moved into eventuate-tram.
Separate module eventuate-tram-optimistic-locking with a EventuateTramOptimisticLockingRetryConfiguration class  to @Import